lib/google_calendar.rb

```diff
@@ -56,14 +61,18 @@ def events_as(who, date)
     @logger.info("Attempting to receive events for "+who)
     @logger.debug details.to_yaml
 
-    ret = @client.execute(
-      api_method: @calendar.events.list,
-      parameters: {'calendarId' => who, 'sendNotifications' => true},
-      body: details.to_json,
-      headers: {'Content-Type' => 'application/json'}
-    )
+    ret = @calendar.list_events(
+        who
+      )
 
     @logger.debug ret.to_yaml
-    ret    
+    ret.to_h  
+  end
+
+  def cleanup!
```

Comment:
Hm. I don't think I understand that snippet of code. Does that change how the GoogleCalendar class gets called? 

Change this part?
 https://github.com/albertsun/huginn/blob/b4fb6e2e276cc8c5908f7d3f8e4257d24b298ca1/app/models/agents/google_calendar_publish_agent.rb#L111  to something like this?

```
calendar_event = GoogleCalendar.new(interpolate_options(options, event), Rails.logger).open |calendar|
   # stuff
   calendar.publish_as( ... )
end
```